### ISSUE TITLE
Execution fails with NRE when using generic field definition

### ISSUE THREAD

**Issue by OlegZee** [fsprojects/FSharp.Data.GraphQL#82]
### Description

I'm defining model with a similar field in several types. E.g.

``` fsharp
let WidgetsField name (getUser: ResolveFieldContext -> 'a -> User) =
    let resolve ctx xx =
        let user = getUser ctx xx 
        let widgets = user.Widgets |> List.toArray
        Connection.ofArray widgets

    Define.Field(name, ConnectionOf Widget, "A person's collection of widgets", Connection.allArgs, resolve)
```

which is instantiated as

``` fsharp
fields = [
        Define.GlobalIdField(fun _ w -> w.Id)
        Define.Field("name", String, fun _ w -> w.Name)
        WidgetsField "widgets" (fun _ user -> user)
    ]
```

The issue is that only the first "widgets" field is functioning. All other instances causes exception in runtime.
### Repro steps

Attaching modifed server.fsx for relay-starter-kit sample project.
- Put this [server.fsx](https://gist.github.com/OlegZee/76181aef0802aabce60dc7dff3ca8fbc) to the sample's folder.
  - Start the server by `fsi server.fsx`.
  - Run the query (in Postman)

```
{"query": "{
    viewer {name}, widgets { edges }
}"}
```

> Notice the first declared "widgets" field is fully functional.
### Actual behavior

NRE exception is raised.

**Comment by Horusiath** [comment]
Could you paste a stack trace?

**Comment by OlegZee** [comment]
I'm sorry for not including such info. It happened in executeField's `let res = info.Definition.Execute innerCtx value` line. `Execute` is null.
Trying to build project and reproduce it on OSX. Will likely provide stack traces soon.

Here's the stack trace:

```
FSharp.Core.dll!Microsoft.FSharp.Core.FSharpFunc<FSharp.Data.GraphQL.Types.ResolveFieldContext, object>.InvokeFast<FSharp.Data.GraphQL.AsyncVal<object>>(Microsoft.FSharp.Core.FSharpFunc<FSharp.Data.GraphQL.Types.ResolveFieldContext, Microsoft.FSharp.Core.FSharpFunc<object, FSharp.Data.GraphQL.AsyncVal<object>>> func, FSharp.Data.GraphQL.Types.ResolveFieldContext arg1, object arg2 = {FSharp.Data.GraphQL.Relay.Connection<******.GraphQL.Schema.WithContext<Persistance.Envelope<DomainTypes.Event>>>})    Unknown
>   FSharp.Data.GraphQL.Server.dll!FSharp.Data.GraphQL.Execution.executeFields@332-1.Invoke(System.Tuple<string, FSharp.Data.GraphQL.Types.ExecutionInfo> tupledArg = {System.Tuple<string,FSharp.Data.GraphQL.Types.ExecutionInfo>}) Line 334  F#
    FSharp.Data.GraphQL.Server.dll!FSharp.Data.GraphQL.Execution.executeFields(FSharp.Data.GraphQL.Types.ObjectDef objdef = {FSharp.Data.GraphQL.Types.ObjectDefinition<FSharp.Data.GraphQL.Relay.Connection<******.GraphQL.Schema.WithContext<Persistance.Envelope<DomainTypes.Event>>>>}, FSharp.Data.GraphQL.Types.ResolveFieldContext ctx = {FSharp.Data.GraphQL.Types.ResolveFieldContext}, object value = {FSharp.Data.GraphQL.Relay.Connection<******.GraphQL.Schema.WithContext<Persistance.Envelope<DomainTypes.Event>>>}, Microsoft.FSharp.Collections.FSharpList<FSharp.Data.GraphQL.Types.ExecutionInfo> fieldInfos = Length = 2) Line 332  F#
    FSharp.Data.GraphQL.Server.dll!FSharp.Data.GraphQL.Execution.createCompletion@201.Invoke(FSharp.Data.GraphQL.Types.ResolveFieldContext ctx = {FSharp.Data.GraphQL.Types.ResolveFieldContext}, object value = {FSharp.Data.GraphQL.Relay.Connection<******.GraphQL.Schema.WithContext<Persistance.Envelope<DomainTypes.Event>>>}) Line 203   F#
    FSharp.Data.GraphQL.Shared.dll!FSharp.Data.GraphQL.AsyncVal.bind@84-1<object, object>.Invoke(object _arg1 = {FSharp.Data.GraphQL.Relay.Connection<******.GraphQL.Schema.WithContext<Persistance.Envelope<DomainTypes.Event>>>}) Line 84 F#
    FSharp.Core.dll!Microsoft.FSharp.Control.AsyncBuilderImpl.args@835-1<object, object>.Invoke(object a)   Unknown
    FSharp.Core.dll!<StartupCode$FSharp-Core>.$Control.loop@430-54(Microsoft.FSharp.Control.Trampoline this = {Microsoft.FSharp.Control.Trampoline}, Microsoft.FSharp.Core.FSharpFunc<Microsoft.FSharp.Core.Unit, Microsoft.FSharp.Control.FakeUnitValue> action)   Unknown
    FSharp.Core.dll!Microsoft.FSharp.Control.Trampoline.ExecuteAction(Microsoft.FSharp.Core.FSharpFunc<Microsoft.FSharp.Core.Unit, Microsoft.FSharp.Control.FakeUnitValue> firstAction) Unknown
    FSharp.Core.dll!Microsoft.FSharp.Control.TrampolineHolder.Protect(Microsoft.FSharp.Core.FSharpFunc<Microsoft.FSharp.Core.Unit, Microsoft.FSharp.Control.FakeUnitValue> firstAction) Unknown
    FSharp.Core.dll!<StartupCode$FSharp-Core>.$Control.-ctor@516-1.Invoke(object state) Unknown
    mscorlib.dll!System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(object state)  Unknown
    mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)   Unknown
    mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)   Unknown
    mscorlib.dll!System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()  Unknown
    mscorlib.dll!System.Threading.ThreadPoolWorkQueue.Dispatch()    Unknown
    mscorlib.dll!System.Threading._ThreadPoolWaitCallback.PerformWaitCallback() Unknown
    [Native to Managed Transition]  
```

**Comment by johnberzy-bazinga** [comment]
@OlegZee @Horusiath 

Until we investigate further, below is a workaround for that issue. 

```
and WidgetConnection = ConnectionOf Widget
and WidgetsField name (getUser: ResolveFieldContext -> 'a -> User) =
    let resolve ctx xx =
        let user = getUser ctx xx 
        let widgets = user.Widgets |> List.toArray
        Connection.ofArray widgets

    Define.Field(name, WidgetConnection, "A person's collection of widgets", Connection.allArgs, resolve)
```

**Comment by OlegZee** [comment]
@johnberzy-bazinga Thank you! This workaround perfectly solves my troubles.

**Comment by OlegZee** [comment]
Put my hands on this project again.

Now getting this NRE on Edge's cursor field:
![fs graphiql nre](https://cloud.githubusercontent.com/assets/513301/19125275/a53a2acc-8b60-11e6-9430-3cf51f6726cd.png)

Schema declaration is as follows and old trick didn't help :)

``` fsharp
    and PostEdge = EdgeOf PostType
    and PostResultPayloadType =
        let gid dbid = toGlobalId "post" (string dbid)
        Define.Object<PostPayload WithContext>(
            name = "PostPayload",
            fields = [
                Define.Field("clientMutationId", Nullable String, justData <| fun p -> p.mutationId)
                Define.Field("id", String,                        justData <| fun pp -> gid pp.envelope.Id)
                Define.Field("postEdge", PostEdge,                dataAndWrap <| fun p wrap -> {Cursor = gid p.envelope.Id; Node = p.envelope |> wrap})
            ])
```

**Comment by johnberzy-bazinga** [comment]
This is fixed with the change made in #101
